Assets/BossRoom/Scripts/Shared/Net/UnityRelayUtilities.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Threading.Tasks;
+using UnityEngine;
+using Unity.Services.Relay;
+using Unity.Services.Relay.Allocations;
+using Unity.Services.Relay.Models;
+
+namespace BossRoom
+{
+    public static class RelayJoinCodeThing
+    {
+        public static string RelayJoinCode = String.Empty;
+    }
+    public class RelayUtility
+    {
+        async public static
+            Task<(string ipv4address, ushort port, byte[] allocationIdBytes, byte[] connectionData, byte[] key, string
+                joinCode)> AllocateRelayServerAndGetJoinCode(int maxConnections, string region = null)
+        {
+            Response<AllocateResponseBody> allocationResponse;
+            Response<JoinCodeResponseBody> createJoinCodeResponse;
+            try
+            {
+                allocationResponse =
+                    await RelayService.AllocationsApiClient.CreateAllocationAsync(
+                        new CreateAllocationRequest(new AllocationRequest(maxConnections)));
+            }
+            catch
+            {
+                Debug.LogError("Relay create allocation request failed");
+                throw;
+            }
+
+            var allocation = allocationResponse.Result.Data.Allocation;
+
+            Debug.Log($"server: {allocation.ConnectionData[0]} {allocation.ConnectionData[1]}");
+            Debug.Log($"server: {allocation.AllocationId}");
+
+            try
+            {
+                createJoinCodeResponse = await RelayService.AllocationsApiClient.CreateJoincodeAsync(
```

Comment:
I know this isn't your code, I'm not sure who to send it to. That api is confusing me. Why can't CreateJoincodeasync just take as an argument an allocationID? Why do we need so many allocations?
Also why the double new?
new CreateJoincodeRequest(new JoinCodeRequest(...

Reply:
This has been updated to use the newer simple API (which was not around in pre.3)